src/adc.rs

```diff
@@ -0,0 +1,562 @@
+//! # Work in Progress
+//! API for the ADC (Analog to Digital Converter)
+//!
+//! Note that the more specific your hardware selection is
+//! (e.g. stm32f303**xc** instead of just stm32f303)
+//! the more functionality is accessible
+//! (in this case: ADC3 and ADC4 additionaly to ADC1 and ADC2).
+//!
+//! # Examples
+//! For a simple, working example check `adc.rs` in the examples folder.
+use cortex_m::asm;
+use embedded_hal::adc::{Channel, OneShot};
+#[cfg(feature = "stm32f303")]
```

Comment:
Okay, not sure what I was seeing, the module is actually gated on "device-selected" now. Still, I think it would make sense to gate it on "stm32f303" and remove those gates here.